Ensure things happen in consistent order

hunse opened the pull request:
- Splitter preserves object order from original network.
- Host/chip receivers are always called in same order.

This makes sure that seeds are always set the same as well. Right now, setting a seed on a network does not actually guarantee it will be the same from one run to the next, because the splitter dictionaries change the order that everything gets built in in a non-deterministic way.

tbekolay commented on nengo_loihi/loihi_cx.py:
```diff
@@ -1164,12 +1166,12 @@ def receive(self, t, x):
     """For receiving host->chip messages"""
 
-    def __init__(self, dimensions, size_out):
+    def __init__(self, dimensions, size_out, **kwargs):
```
I think `**kwargs` is confusing here; for me, I don't really remember what other parameters there are to `Node`. If I look it up and see `size_in`, I'll get a `TypeError` if I try to specify it. I think we should bite the bullet and expose `label` and `seed` (or just `label` if seeds are irrelevant here).

tbekolay commented on nengo_loihi/loihi_cx.py:
```diff
@@ -1190,6 +1192,7 @@ def collect_spikes(self):
 class ChipReceiveNeurons(ChipReceiveNode):
     """Passes spikes directly (no on-off neuron encoding)"""
-    def __init__(self, dimensions, neuron_type=None):
+    def __init__(self, dimensions, neuron_type=None, **kwargs):
```
Here also, explicitly include `label` and (perhaps) `seed`.

hunse commented:
I also can't think of a test that is non-trivial. Maybe #70 will test some of this, though also the point of #70 was to make sure that things have the same seeds as Nengo no matter what order they're built in.

I've addressed the comments and added a changelog.

tbekolay commented:
From the review: we should add a test that splits a model multiple times and ensures that their order in the split networks is the same.

hunse commented:
I will write this test.

hunse commented:
Ok, I've added a test. I've tested that it can distinguish between before this PR and after it.
